app/assets/javascripts/float-kit/addon/modifiers/apply-floating-ui.js

```diff
@@ -0,0 +1,110 @@
+import Modifier from "ember-modifier";
+import { registerDestructor } from "@ember/destroyable";
+import { bind } from "discourse-common/utils/decorators";
+import {
+  arrow,
+  autoPlacement,
+  autoUpdate,
+  computePosition,
+  inline,
+  offset,
+  shift,
+} from "@floating-ui/dom";
+import { headerOffset } from "discourse/lib/offset-calculator";
+import { iconHTML } from "discourse-common/lib/icon-library";
+import { cached } from "@glimmer/tracking";
+import domFromString from "discourse-common/lib/dom-from-string";
+
+export default class FloatKitApplyFloatingUi extends Modifier {
+  constructor(owner, args) {
+    super(owner, args);
+    registerDestructor(this, (instance) => instance.cleanup());
+  }
+
+  modify(element, [trigger, options]) {
+    this.element = element;
+    this.trigger = trigger;
+    this.options = options ?? {};
+
+    if (this.options.arrow) {
+      element.appendChild(this.arrowElement);
+    }
+
+    this.cleanup = autoUpdate(trigger, element, this.update);
+  }
+
+  @cached
+  get arrowElement() {
+    return domFromString(
+      iconHTML("tippy-rounded-arrow", { class: "arrow" })
+    )[0];
+  }
+
+  @bind
+  async update() {
+    const middleware = [
+      offset(this.options.offset ? parseInt(this.options.offset, 10) : 10),
+      autoPlacement({
+        allowedPlacements: (this.options.allowedPlacements ?? "top|bottom")
+          .split("|")
+          .filter(Boolean),
+        padding: this.options.padding ?? { top: headerOffset() },
+      }),
+      shift(),
+    ];
+
+    if (this.options.inline) {
+      middleware.push(inline());
+    }
+
+    if (this.options.arrow) {
+      middleware.push(arrow({ element: this.arrowElement }));
+    }
+
+    // {
+    //   getBoundingClientRect: () => {
+    //     return this.trigger.range.getBoundingClientRect();
+    //   },
+    //   getClientRects: () => {
+    //     return this.trigger.range.getClientRects();
+    //   },
+    // },
+
+    this.element.dataset.strategy = this.options.strategy || "absolute";
+
+    await computePosition(this.trigger, this.element, {
+      strategy: this.options.strategy || "absolute",
+      middleware,
+    }).then(({ x, y, placement, middlewareData }) => {
```

Comment:
Using `await` and also `.then()` seems a bit weird. Could we remove the `.then` and just use `await`?